src/rules/rdf/mustNotExist.js

```diff
@@ -0,0 +1,57 @@
+import { RDF_OBSOLETE_TAGS, RDF_UNALLOWED_TAGS, RDF_UNALLOWED_IF_LISTED_TAGS,
+  VALIDATION_ERROR, VALIDATION_WARNING } from 'const';
+import * as messages from 'messages';
+
+
+export var mustNotExist = (xmlDoc, namespace, filename=null) => {
+  return new Promise((resolve) => {
+    var bannedTags = RDF_UNALLOWED_TAGS;
+    var validatorMessages = [];
+
+    var addonIsListed = xmlDoc
+      .getElementsByTagNameNS(namespace, 'listed').length > 0;
+
+    if (addonIsListed) {
+      bannedTags = bannedTags.concat(RDF_UNALLOWED_IF_LISTED_TAGS);
+    }
+
+    // Using any banned tag is an error.
+    validatorMessages = validatorMessages.concat(
+      _checkForTags(xmlDoc, namespace, bannedTags, VALIDATION_ERROR,
+                    'TAG_NOT_ALLOWED_', filename));
+
+    // But using an obsolete tag is just a warning.
+    validatorMessages = validatorMessages.concat(
+      _checkForTags(xmlDoc, namespace, RDF_OBSOLETE_TAGS, VALIDATION_WARNING,
+                    'TAG_OBSOLETE_', filename));
+
+    resolve(validatorMessages);
+  });
+};
+
+export var _checkForTags = (
+(xmlDoc, namespace, tags, severity, errorCodePrefix, filename=null) => {
```

Comment:
Do you need the fat arrow function? 

Also perhaps use an object for the args - that will make it clearer in the caller what args you're passing.

e.g. 

```
function _checkForTags({xmlDoc, namespace, tags, severity, errorCodePrefix, filename=null} = {}) {
  // use any arg passed in the object directly here.
}

_checkForTags({severity: 'hai'});

```

Here's an example showing that sort of thing in action:

http://jsbin.com/cowaki/edit?js,console